apollo-router/src/services/layers/persisted_queries/manifest_poller.rs

```diff
@@ -347,23 +347,32 @@ async fn poll_uplink(
     mut drop_receiver: mpsc::Receiver<()>,
     http_client: Client,
 ) {
+    let http_client = http_client.clone();
     let mut uplink_executor = stream::select_all(vec![
-        stream_from_uplink::<PersistedQueriesManifestQuery, MaybePersistedQueriesManifestChunks>(
-            uplink_config.clone(),
-        )
-        .filter_map(|res| {
+        stream_from_uplink_transforming_new_response::<
+            PersistedQueriesManifestQuery,
+            MaybePersistedQueriesManifestChunks,
+            Option<PersistedQueryManifest>,
+        >(uplink_config.clone(), move |response| {
             let http_client = http_client.clone();
-            let graph_ref = uplink_config.apollo_graph_ref.clone();
-            async move {
-                match res {
-                    Ok(Some(chunks)) => match manifest_from_chunks(chunks, http_client).await {
-                        Ok(new_manifest) => Some(ManifestPollEvent::NewManifest(new_manifest)),
-                        Err(e) => Some(ManifestPollEvent::FetchError(e)),
-                    },
-                    Ok(None) => Some(ManifestPollEvent::NoPersistedQueryList { graph_ref }),
-                    Err(e) => Some(ManifestPollEvent::Err(e.into())),
+            Box::new(Box::pin(async move {
```

Comment:
I'm wondering if we can't just keep the `Box::pin` and get rid of the `Box::new`

Reply:
Doesn't look like it (similar error to the other one)